### PR TITLE
Add details option for GET /containers/{id}/logs

### PR DESCRIPTION
**Pull request by yuting-liu**
In page `Engine API version history` (https://docs.docker.com/engine/api/version-history/), there is a new `v1.23 API change` for `GET /containers/(id or name)/logs`, which now accepts a `details` query parameter to stream the extra attributes that were provided to the containers `LogOpts`, such as environment variables and labels, with the logs.
This `details` is a boolean option. If `true`, extra details provided to logs will be shown. The default value is `false`.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/967)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java [hunk 1]
```diff
@@ -197,4 +198,54 @@ public void asyncLogContainerWithSince() throws Exception {
 
         assertThat(loggingCallback.toString(), containsString(snippet));
     }
+
+    @Test
+    public void asyncLogContainerWithDetails() throws Exception {
+        CreateContainerResponse container = dockerRule.getClient().createContainerCmd("busybox")
+                .withCmd("--log-driver", "journald", "--log-opt", "env=ENV_VAR", "-e", "ENV_VAR=logtester.1234", "flyinprogrammer/logtester")
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java]: how does this work?

Comment (yuting-liu) [reply src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java]: If run with docker, the command should be `docker run --log-driver journald --log-opt env=SOME_ENV_VAR -e SOME_ENV_VAR=logtester.1234 flyinprogrammer/logtester`. I'm not sure how to set it with `.withCmd`.

Comment (KostyaSha) [reply src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java]: CMD is a CMD for binary
What you did is `docker run busybox sh --log-driver journald --log-opt env=SOME_ENV_VAR -e SOME_ENV_VAR=logtester.1234 flyinprogrammer/logtester`
createContainer() is a java command 
createContainer().withDetails() ? Look at examples in existing test case.

File: src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java [hunk 2]
```diff
@@ -201,8 +203,12 @@ public void asyncLogContainerWithSince() throws Exception {
 
     @Test
     public void asyncLogContainerWithDetails() throws Exception {
-        CreateContainerResponse container = dockerRule.getClient().createContainerCmd("busybox")
-                .withCmd("--log-driver", "journald", "--log-opt", "env=ENV_VAR", "-e", "ENV_VAR=logtester.1234", "flyinprogrammer/logtester")
+        HashMap<String, String> config = new HashMap<>();
+        config.put("env", "ENV_VAR");
+        LogConfig logConfig = new LogConfig(LogConfig.LoggingType.JOURNALD, config);
+        CreateContainerResponse container = dockerRule.getClient().createContainerCmd("flyinprogrammer/logtester")
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/cmd/LogContainerCmdIT.java]: please use busybox for tests

**Pull request closed without merging**